### PR TITLE
Add latest version of Ninja

### PR DESCRIPTION
**Pull request by adamjstewart**
Also adds build tests to `ninja` and `ninja-fortran`.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ninja-fortran/package.py [hunk 1]
```diff
@@ -23,27 +23,36 @@
...
-    """ A Fortran capable fork of ninja """
+    """A Fortran capable fork of ninja."""
+
     homepage = "https://github.com/Kitware/ninja"
     url      = "https://github.com/Kitware/ninja/archive/v1.7.2.gcc0ea.kitware.dyndep-1.tar.gz"
 
     version('1.7.2', '3982f508c415c0abaca34cb5e92e711a')
 
-    extends('python')
+    depends_on('python', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ninja-fortran/package.py]: Doesn't build any Python libraries, therefore `depends_on` instead of `extends`.

**Comment by aprokop** [comment]
@adamjstewart Just noticed the request, I'll take a look later today.

**Comment by adamjstewart** [comment]
@aprokop What is `ninja-build`? The installation process doesn't create an executable called `ninja-build`, nor does the [Homebrew recipe](https://github.com/Homebrew/homebrew-core/blob/master/Formula/ninja.rb). Just wondering why it's necessary.

**Comment by aprokop** [comment]
@adamjstewart In some Linux distrox (e.g., Fedora) when you install the system ninja package (called `ninja-build`) you get `ninja-build` but not `ninja`.

**Comment by adamjstewart** [comment]
@aprokop Fixed, thanks for the explanation!

It looks like there was also a new `ninja-fortran` release. Added that too.

**Pull request merged**